Remove support for expr_form

rallytime opened the pull request:
This option has been deprecated and marked for removal in Fluorine.

``tgt_type`` should be used instead.

Fixes #46467

Commit:
Remove support for expr_form

This option has been deprecated and marked for removal in Fluorine.

``tgt_type`` should be used instead.

cachedout commented on doc/topics/releases/fluorine.rst:
```diff
@@ -27,6 +27,19 @@ syndic respects :conf_minion:`enable_legacy_startup_events` as well.
+
+Those who are using the :ref:`LocalClient <local-client>` (either directly,
+or implicitly via a :ref:`netapi module <all-netapi-modules>`) need to update
+their code to use ``tgt_type``.
```
I think we could do with an example here.

rallytime replied:
Sure thing.

cachedout commented on salt/runners/cache.py:
```diff
@@ -95,7 +76,7 @@ def pillar(tgt=None, tgt_type='glob', **kwargs):
 
 
-def mine(tgt=None, tgt_type='glob', **kwargs):
+def mine(tgt=None, tgt_type='glob'):
```
Are we quite certain nothing else is coming in through kwargs here?

rallytime replied:
I was waffling on this one, because `kwargs` isn't used in this function, but after looking at the test results I think removing this is premature. I'll add it back in.

Commit:
Add **kwargs back into mine function in cache runner

This was removed during the deprecation removal process and shouldn't have been.

Commit:
Add example to release notes about tgt_type usage

Commit:
Add **kwargs back into grains and pillar funcs in cache runner

This was removed during the deprecation removal process and shouldn't have been.
